File: backend/app/schemas/event.py
from pydantic import BaseModel


class RailEventNode(BaseModel):
	id: str
	type: str  # 'line_failure' | 'derailment' | 'speed_restriction' | 'signal_failure'
	severity: str  # 'minor' | 'major'
	status: str  # 'active' | 'resolved'
	segmentId: str | None = None
	fromStationId: str | None = None
	toStationId: str | None = None
	stationId: str | None = None
	trainId: str | None = None
	restrictedVmax: int | None = None
	message: str
	startedAt: float
	resolvesAt: float
	resolvedAt: float | None = None


class EventsSnapshotResponse(BaseModel):
	events: list[RailEventNode]
	timestamp: float


class IncidentCreateRequest(BaseModel):
	type: str  # 'line_failure' | 'derailment' | 'speed_restriction' | 'signal_failure'
	# Zależnie od typu: id odcinka (line_failure/speed_restriction), pociągu
	# (derailment) albo stacji (signal_failure) — wskazany wprost przez użytkownika.
	targetId: str

File: frontend/src/lib/types/event.ts
export type RailEventType = 'line_failure' | 'derailment' | 'speed_restriction' | 'signal_failure';
export type RailEventSeverity = 'minor' | 'major';
export type RailEventStatus = 'active' | 'resolved';

export type RailEventNode = {
	id: string;
	type: RailEventType;
	severity: RailEventSeverity;
	status: RailEventStatus;
	segmentId: string | null;
	fromStationId: string | null;
	toStationId: string | null;
	stationId: string | null;
	trainId: string | null;
	restrictedVmax: number | null;
	message: string;
	startedAt: number;
	resolvesAt: number;
	resolvedAt: number | null;
};

export type EventsSnapshotResponse = {
	events: RailEventNode[];
	timestamp: number;
};
